lib/spack/spack/build_environment.py

```diff
@@ -141,7 +141,6 @@ def clean_environment():
     # can affect how some packages find libraries.  We want to make
     # sure that builds never pull in unintended external dependencies.
     env.unset('LD_LIBRARY_PATH')
-    env.unset('CRAY_LD_LIBRARY_PATH')
```

Comment:
This is the only part of this PR that scares me, I'm worried there might be a reason we need to clean it.